server/src/main/java/org/apache/cassandra/sidecar/db/schema/SystemAuthSchema.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.db.schema;
+
+import com.datastax.driver.core.KeyspaceMetadata;
+import com.datastax.driver.core.Metadata;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.Session;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Schema for getting information stored in system_auth keyspace.
+ */
+public class SystemAuthSchema extends TableSchema
+{
+    private static final String IDENTITY_TO_ROLE_TABLE = "identity_to_role";
+    PreparedStatement selectRoleFromIdentity;
+    PreparedStatement getAllRolesAndIdentities;
+
+    protected String keyspaceName()
+    {
+        return "system_auth";
+    }
+
+    @Override
+    protected void prepareStatements(@NotNull Session session)
+    {
+        selectRoleFromIdentity = prepare(selectRoleFromIdentity,
+                                          session,
+                                          CqlLiterals.selectRoleFromIdentity());
+
+        getAllRolesAndIdentities = prepare(getAllRolesAndIdentities,
+                                           session,
+                                           CqlLiterals.getAllRolesAndIdentities());
+    }
+
+    protected String tableName()
+    {
+        throw new UnsupportedOperationException("SystemAuthSchema supports reading information from multiple " +
+                                                "tables in system_auth keyspace");
+    }
+
+    @Override
+    protected boolean exists(@NotNull Metadata metadata)
```

Comment:
Is there integration test for the scenario when connecting to cassandra version, say 4.0, and the table does not exist.

Reply:
test ` testAssociatingRoleWithNonAdminIdentity` in `MutualTlsAuthenticationIntegrationTest` tests this. in 4.x cluster we expect a failure if the identity is not an admin identity (when table does not exists, it allows only admin identities). In 5.x cluster, if the identity is part of the table and not an admin identity we allow it through.